packages/cli/src/core/builder/hardhat.ts

```diff
@@ -87,19 +87,24 @@ export class HardhatBuilder extends BaseBuilder {
             !info.evm ||
             !info.evm.bytecode ||
             !info.evm.bytecode.object ||
-            !info.metadata ||
             !info.abi
           ) {
             logger.debug("Skipping", contractPath, "(no bytecode or metadata)");
             continue;
           }
 
           const bytecode = info.evm.bytecode.object;
-          const deployedBytecode = info.evm.deployedBytecode.object;
-          const { metadata, abi } = info;
+          const deployedBytecode = info.evm.deployedBytecode
```

Comment:
what's the explanation here? deployedBytecode should always be there no?

Reply:
was trying this out for alt chains which have different structure for their compiled artifacts

Reply:
closing this PR for now